code/web/src/modules/profileForm/ProfileForm.js

```diff
@@ -20,19 +21,27 @@ class ProfileForm extends Component{
         description: ''
       }
   }
+  componentDidMount() {
+    this.props.getListByUser()
+  }
 
   updateProfileState(e) {
     this.setState({ [e.target.name]: e.target.value })
   }
 
   updateProfile = () => {
     let newState = {
-          img: this.state.img || this.props.user.details.img || '',
+          id: this.props.user.details.id,
+          image: this.state.img || this.props.user.details.img || '',
           address: this.state.address || this.props.user.details.address || '',
           email: this.state.email || this.props.user.details.email || '',
           description: this.state.description || this.props.user.details.description || ''
         }
-    this.props.saveProfile(newState)
+    let newDate = {
+      id: this.props.subscriptionsByUser.list[0].id,
+      nextDeliveryDate: this.props.subscriptionsByUser.nextDeliveryDate
+    }
+    this.props.saveProfile(newState, newDate, this.props.subscriptionsByUser.list.map(x => x.id))
```

Comment:
I'd like to explain what's going on in lines 40 - 44 just to make sure I'm understanding correctly.

We've set a newDate object so that `nextDeliveryDate` can be updated in the table on the backend.  
We're passing newState (as before), newDate (to send the post request, because Redux store is being updated onChange with `updateAvailability`), and the `id` of each crate in the list will be sent in an array as the third argument for `this.props.saveProfile()` ?

Reply:
Yeah I should have explained this one, thanks for summing it up! The map should probably be placed in a variable for readability, then passed in `saveProfile`. I made the decision to send the `nextDeliverDate` on save to avoid making tons of requests on chance.